Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -291,24 +351,23 @@ def L(self, value):
             raise TypeError('The Lipschitz constant is a real positive number')
 
     def calculate_Lipschitz(self):
+        r""" Default value for the Lipschitz constant."""
+        
         # Compute the Lipschitz parameter from the operator if possible
         # Leave it initialised to None otherwise
         self._L = (1./self.gradient.norm())**2  
     
     @property
     def gradient(self):
-        '''creates a gradient operator if not instantiated yet
+        r""" GradientOperator is created if it is not instantiated yet. The domain of the `_gradient`,
+        is created in the `__call__` and `proximal` methods. 
 
-        There is no check that the variable _domain is changed after instantiation (should not be the case)'''
-        if self._gradient is None:
-            if self._domain is not None:
-                self._gradient = GradientOperator(self._domain, correlation = self.correlation, backend = self.backend)
+        """
+        if self._domain is not None:
+            self._gradient = GradientOperator(self._domain, correlation = self.correlation, backend = self.backend)
         return self._gradient
```

Comment:
Are we happy to return `None` if `self._domain` is not set?
IMHO we should give a better error message and raise an Error in this method.

Reply:
At the instance level `self._domain`  is `None`. It will remain `None` if we do not use any of the `methods` of this class. The same applies for the `FGP_TV` class where there is no `_domain` attribute. It will not be `None` if either the `call` or the `proximal` method are called. Since we do not have a `domain` attribute for our `Function` I think that the right value should be `None`.

Reply:
Fixed